url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/net/http/SimpleHttpServer.java

```diff
@@ -0,0 +1,113 @@
+package net.hetimatan.net.http;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+import net.hetimatan.io.file.KyoroFile;
+import net.hetimatan.io.file.KyoroFileForFiles;
+import net.hetimatan.io.filen.ByteKyoroFile;
+import net.hetimatan.io.filen.CashKyoroFileHelper;
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.net.http.HttpServerFront;
+import net.hetimatan.util.event.net.io.KyoroSocket;
+import net.hetimatan.util.http.HttpObjectHelper;
+import net.hetimatan.util.http.HttpRequest;
+
+//@todo
+public class SimpleHttpServer extends HttpServerTemplate {
+
+	private CashKyoroFile mFile = null;
+	private LinkedHashMap<String, String> mMimetype = new LinkedHashMap<String,String>();
+	{
+		mMimetype.put("mp4", "video/mp4");
+		mMimetype.put("txt", "text/plain");
+		mMimetype.put("html", "text/html");
+	}
+
+	public SimpleHttpServer() { 
+		try {
+			mFile = new CashKyoroFile(new File("../../h264.mp4"), 16*1024, 4);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *mFile*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).